ci/cpu/clang12_release.yml

```diff
@@ -0,0 +1,40 @@
+include:
+  - local: 'ci/common-ci.yml'
+
+cpu clang12 release deps:
+  extends: .build_deps_common
+  variables:
+    DOCKERFILE: ci/docker/build.Dockerfile
+    BASE_IMAGE: docker.io/ubuntu:22.04
+    EXTRA_APTGET: "clang-12 libomp-12-dev"
+    COMPILER: clang@12.0.1
+    CXXSTD: 17
+    USE_MKL: "ON"
+    USE_ROCBLAS: "OFF"
+    SPACK_ENVIRONMENT: ci/docker/release-cpu.yaml
+    COMMON_SPACK_ENVIRONMENT: ci/docker/common.yaml
+    BUILD_IMAGE: $CSCS_REGISTRY_PATH/release-cpu-clang12/build
+    USE_CODECOV: "false"
+
+# Builds a Docker image for the current commit
+cpu clang12 release build:
+  extends: .build_common
+  needs:
+    - cpu clang12 release deps
+  variables:
+    DOCKERFILE: ci/docker/deploy.Dockerfile
+    DEPLOY_BASE_IMAGE: docker.io/ubuntu:22.04
+    EXTRA_APTGET_DEPLOY: "glibc-tools jq strace"
+    # glibc-tools is needed for libSegFault on ubuntu:22.04
+    DEPLOY_IMAGE: $CSCS_REGISTRY_PATH/release-cpu-clang12/deploy:$CI_COMMIT_SHA
+    SLURM_CONSTRAINT: mc
+    THREADS_PER_NODE: 72
```

Comment:
I guess we could move a few more variables up to `build_*common`:
- `DOCKERFILE`
- `BASE_IMAGE`
- `COMMON_SPACK_ENVIRONMENT`
assuming it's possible to override variables when needed (which it seems like it is). Things like `CXXSTD` could at least default to `17` and `USE_ROCBLAS` to `OFF`.

Would it also make sense to define a few reusable blocks like `mc_constraint` and `gpu_constraint`? As far as I can tell it's possible to use multiple base configurations in `.extends` (https://docs.gitlab.com/ee/ci/yaml/yaml_optimization.html#merge-details). Obviously a "reusable base configuration" with only one variable doesn't make much sense, so this is only useful if we can group more than one variable.

I admit I haven't tried this so I'm not 100% sure it works.

Reply:
Good idea. Let me try.